subsys/debug/CMakeLists.txt

```diff
@@ -5,4 +5,9 @@ zephyr_sources_ifdef(
   openocd.c
   )
 
+zephyr_sources_ifdef(
```

Comment:
Nit: This file is so posix arch dependent that I would only include if ARCH_POSIX

Reply:
Actually `CONFIG_ASAN` already takes this in account through Kconfig `depends on`.
So I rather keep it a simple here; just in case the `depends on` changes in the Kconfig, else we need to update it in 2 places.

Reply:
Fair enough